### ISSUE TITLE
Make direction arrows more visible

### ISSUE THREAD

**Issue by tordans** [openstreetmap/iD#8858]
This first description is outdated, see comment below.

---

### URL

https://www.openstreetmap.org/way/4068043

### How to reproduce the issue?

The `highway=primary` does not who any direction arrows. However, I need those to see if the `sidepath=right` on the object is correct.

Edit: https://www.openstreetmap.org/edit?way=4068043#map=18/52.49599/13.46307

### Screenshot(s) or anything else?

![image](https://user-images.githubusercontent.com/111561/146736548-318b2dab-2de7-4cc3-be20-88f5445ba4b5.png)

### Which iD Editor versions do you see the issue on?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Chrome

**Comment by LevinGar** [comment]
Hi Tordans! The street Elsenbrücke seems to be both directions when it is on the bridge, both directions of this street intersects starting the bridge and separates again on Elsenbrücke. As long as I know, arrows are only shown on one way streets.
Additionally one sidepath is a cyclepath which has a one way specification, and the other is a sidewalk which has not which seems to be correct.
It is expected to have the arrows over the bridge considering that is not a one way street?

**Comment by tordans** [comment]
You are right @LevinGar, thanks for making me look at this again. My issue is more nuanced: The direction arrows are not visible enough in some situations (and only sometimes fully hidden).

**Issue A:**

On some way segments, there is only one direction arrow for a way that spans my whole screen height.
<img width="627" alt="Bildschirmfoto 2021-12-22 um 21 39 06" src="https://user-images.githubusercontent.com/111561/147153451-28269cee-9a6a-4849-a25a-77467f6c0609.png">

Which is too few IMO, and not the same behavior everywhere. For example this path nearby has two arrows on a lot smaller line.
<img width="320" alt="Bildschirmfoto 2021-12-22 um 21 38 37" src="https://user-images.githubusercontent.com/111561/147153515-37f6567b-9aae-45ec-8970-66493509006f.png">

=> Suggestion: Make it more arrows; the same distance for all ways.

**Issue B:** 

And sometimes there is no arrow at all. But to reproduce this, the screen height and map position has to be just right. But there are also other things involved, because once I moved the screen and then try to reproduce, there are arrows showing up.

URL: https://www.openstreetmap.org/edit?way=4068043#map=19/52.49576/13.46286
Screenshot:
<img width="609" alt="Bildschirmfoto 2021-12-22 um 21 39 36" src="https://user-images.githubusercontent.com/111561/147153057-8db7995f-2a1c-4b6d-8dcc-b62957559042.png">

=> Suggestion: Cannot pinpoint this bug, yet.

**Issue C:**

When the label overlaps the (one) direction arrow, then this arrow is only shown when (a) the way is selected and (b) the mouse hovers the label name (so the name gets hidden). This causes even more situations, it looks like there is no direction arrow at all.

Here is a screencast video: https://share.icloud.com/photos/0b6g329Ads98fQIHkc8qXrmnw

=> Suggestion: Solving Issue A would make this less of a problem. Otherwise, hide the name whenever I hover the any part of the road. Or, move the arrow so it never is placed below the name.

**Issue D:**

I wonder why we have a different behavior for hovered and selected ways. In both cases, the name disappears and the highlight-glow shows, but only on selected ways the arrows are visible. 

=> Suggestion: Make hover and selected the same, both should show the arrows.

**Comment by tyrasd** [comment]
> Issue A:
> Make it more arrows

the issue here is that the small triangles perform double-duty: it's a draggable virtual vertex to be able to add new nodes, as well as an indicator for the direction of the way.

Adding more triangles would also mean more virtual vertices to drag the node. I'm not sure if that's what we want…

Or, we split this up into multiple markers for the direction of the way, and a separate one for the draggable vertex. 🤔 

See also #12099 for an alternative approach to indicate the "sides" of the selected way. -> maybe that's the most approachable way forward for the moment: it does not require introducing different markers _on the way_ itself that can be confused with the draggable vertex, and work well for the left/right tags that we current have fields for. Only for potential new fields that work on forward/backward (e.g. maxspeed:*) we'd need to find a way to extend that mechanism.

> Issue B:

see above

> Issue D:
> Make hover and selected the same, both should show the arrows.

This might makes sense, but as mentioned above: As the triangles have this double-duty job at the moment, it's could lead to confusion if mappers interpret them as a draggable pseudo-vertex.


> Issue C:

the easiest solution would be to omit the name when the feature is selected, as the name is then already visible on the field in the sidebar.
